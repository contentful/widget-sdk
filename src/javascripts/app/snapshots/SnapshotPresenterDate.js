import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import { userInputFromDatetime } from './dateUtils';

const MODE_DATE = 'date';
const MODE_TIMEZ = 'timeZ';

const SnapshotPresenterDate = ({ settings, value }) => {
  const { date, time, utcOffset } = userInputFromDatetime(value);
  const mode = settings.format || MODE_DATE;

  let dateString = moment(date).format('dddd, MMMM Do YYYY');
  if (mode === MODE_DATE) return <span data-test-id="snapshot-presenter-date">{dateString}</span>;

  const is24HourTimezone = parseInt(settings.ampm, 10) === 24;
  if (is24HourTimezone) {
    dateString += `, ${time}`;
  } else {
    const [hour, minute] = time.split(':');
    const longDate = moment().hour(hour).minute(minute).format('LT');
    dateString += `, ${longDate}`;
  }

  if (mode === MODE_TIMEZ) {
    dateString += `, UTC${utcOffset}`;
  }

  return <span data-test-id="snapshot-presenter-date">{dateString}</span>;
};

SnapshotPresenterDate.propTypes = {
  settings: PropTypes.shape({
    format: PropTypes.string,
    ampm: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  }).isRequired,
  value: PropTypes.string.isRequired,
};

SnapshotPresenterDate.defaultProps = {
  settings: {},
};

export default SnapshotPresenterDate;
