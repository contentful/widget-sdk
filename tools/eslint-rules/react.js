module.exports = {
  extends: ['plugin:react/recommended', 'prettier/react'],
  plugins: ['react-hooks'],
  settings: {
    react: {
      version: '16.10.2',
    },
  },
  rules: {
    'react-hooks/rules-of-hooks': 'error',
    'react-hooks/exhaustive-deps': 'error',
    'react/display-name': 'off',
    'react/prop-types': ['error', { ignore: ['children'] }],
    'react/no-unused-prop-types': 'error',
    'react/jsx-no-target-blank': ['error', { enforceDynamicLinks: 'always' }],
  },
};
