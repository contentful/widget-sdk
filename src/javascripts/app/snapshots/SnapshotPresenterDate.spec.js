import React from 'react';
import { render } from '@testing-library/react';
import SnapshotPresenterDate from './SnapshotPresenterDate';

describe('SnapshotPresenterDate', () => {
  it('should render standard 24H string', () => {
    const { container } = render(
      <SnapshotPresenterDate
        settings={{ format: 'timeZ', ampm: 24 }}
        value="2020-03-10T16:00+00:00"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render standard AmPm string', () => {
    const { container } = render(
      <SnapshotPresenterDate
        settings={{ format: 'time', ampm: 12 }}
        value="2020-03-10T16:00+00:00"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should render default string', () => {
    const { container } = render(
      <SnapshotPresenterDate settings={{}} value="2020-03-10T16:00+00:00" />
    );
    expect(container).toMatchSnapshot();
  });
});
