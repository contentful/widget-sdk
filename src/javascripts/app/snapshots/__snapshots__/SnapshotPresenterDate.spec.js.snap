// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapshotPresenterDate should render default string 1`] = `
<div>
  <span>
    Tuesday, March 10th 2020
  </span>
</div>
`;

exports[`SnapshotPresenterDate should render standard 24H string 1`] = `
<div>
  <span>
    Tuesday, March 10th 2020, 16:00, UTC+00:00
  </span>
</div>
`;

exports[`SnapshotPresenterDate should render standard AmPm string 1`] = `
<div>
  <span>
    Tuesday, March 10th 2020, 4:00 PM
  </span>
</div>
`;
